src/services/AdminApiContext.ts

```diff
@@ -123,31 +123,50 @@ export class AdminApiContext {
     }
 
     isAuthenticated(): boolean {
-        // TODO: check token expiry
-        return !!this.options['access_token'];
+        if (!this.options['access_token']) {
+            return false;
+        }
+
+        const token = this.options['access_token'];
+        const decodedToken: { exp: number } = jwtDecode(token);
+        const currentTime = Math.floor(Date.now() / 1000);
+        return decodedToken.exp > currentTime;
+    }
+
+    async ensureAuthenticated(): Promise<void> {
+        if (!this.isAuthenticated()) {
+            this.options['access_token'] = await AdminApiContext.authenticateWithClientCredentials(this.context, this.options);
+            this.context = await AdminApiContext.createApiRequestContext(this.options);
+        }
     }
 
     async get<PAYLOAD>(url: string, options?: RequestOptions<PAYLOAD>): Promise<APIResponse> {
+        await this.ensureAuthenticated();
```

Comment:
Maybe another solution to consider: Instead of decoding the token locally and comparing the expiration date every time, we could just wait for the response here, and if the response is an unauthorized response code, we could fetch a new token and rerun the request. That is how the administration does it, if I am not mistaken.